posts/Stef/2019-11-18-hibernate-orm-with-panache-in-quarkus.adoc

```diff
@@ -0,0 +1,465 @@
+    }
+    
+    public List<Person> findBornAfter(Date date){
+        return find("birth > :date", Parameters.with("birth", date)).list();
+    }
+}
+----
+
+Note that the `find` convenience method allows HQL, but also _simplified_ HQL (think of it as contextualized HQL):
+
+- If your query is empty, it expands to `FROM <entityType>`
+- If your query starts with `FROM` or `SELECT`, it is left alone as HQL
+- If your query starts with `ORDER BY...` it expands to `FROM <entityType> ORDER BY...`
+- If your query only has a single property and argument, it expands to `FROM <entityType> WHERE <property> = <argument>`
+- Otherwise, your query is taken as a `WHERE...` clause and expands to `FROM <entityType> WHERE...`
+
+This allows many simple queries to be simplified to a minimum, while allowing complex queries to be left as-is.
+
+Now, our REST endpoint is not changed much, but let's include it for good measure:
+
+[source%nowrap,java]
+----
+@Path("/")
+public class PersonEndpoint {
+    @Inject
+    private PersonDao personDao;
+    
+    @GET
+    @Path("people")
+    public List<Person> all(){
+        return personDao.findAll().list();
+    }
+
+    @GET
+    @Path("people/by-name")
+    public List<Person> findByName(@PathParam String name){
+        return personDao.findByName(name);
+    }
+
+    @GET
+    @Path("people/born-after")
+    public List<Person> findBornAfter(@PathParam Date date){
+        return personDao.findBornAfter(date);
+    }
+
+    @GET
+    @Path("person/{id}")
+    public Person findById(@PathParam Long id){
+        Person p = personDao.findById(id);
+        if(p == null)
+            throw new WebApplicationException(Status.NOT_FOUND);
+        return p;
+    }
+
+    @PUT
+    @Path("person/{id}")
+    public void updatePerson(@PathParam Long id, Person newPerson){
+        Person p = personDao.findById(id);
+        if(p == null)
+            throw new WebApplicationException(Status.NOT_FOUND);
+        p.birth = newPerson.birth;
+        p.firstName = newPerson.firstName;
+        p.lastName = newPerson.lastName;
+    }
+
+    @DELETE
+    @Path("person/{id}")
+    public void deletePerson(@PathParam Long id){
+        Person p = personDao.findById(id);
+        if(p == null)
+            throw new WebApplicationException(Status.NOT_FOUND);
+        personDao.delete(p);
+    }
+    
+    @POST
+    @Path("people")
+    public Response newPerson(@Context UriInfo uriInfo, Person newPerson){
+        Person p = new Person();
+        p.birth = newPerson.birth;
+        p.firstName = newPerson.firstName;
+        p.lastName = newPerson.lastName;
+        personDao.persist(p);
+        
+        URI uri = uriInfo.getAbsolutePathBuilder()
+                .path(PersonEndpoint.class)
+                .path(PersonEndpoint.class, "findById")
+                .build(p.id);
+        return Response.created(uri).build();
+    }
+}
+----
+
+This is already a long way towards reducing boilerplate, don't you think?
```

Comment:
This statement seems a bit out of place right after we showed the REST endpoint which didn't change much. Maybe move it up to line 255 which is right after where we show the simplified `PersonDao`?

Reply:
yes